sess.go

```diff
@@ -91,41 +92,42 @@ func (s *session) Serve() error {
 
 	// query-cycle
 	for {
-		msg, err = t.Read()
+		msg, err := t.NextFrontendMessage()
 		if err != nil {
 			return err
 		}
 
-		switch msg.Type() {
-		case protocol.Terminate:
+		switch msg.(type) {
+		case *pgproto3.Terminate:
 			s.Conn.Close()
 			return nil // client terminated intentionally
-		case protocol.Query:
-			sql, err := msg.QueryText()
-			if err != nil {
-				return err
+		case *pgproto3.Query:
+			q := &query{
+				transport: t,
+				sql:       msg.(*pgproto3.Query).String,
+				queryer:   s.Server,
+				execer:    s.Server,
 			}
-			q := &query{transport: t, sql: sql, queryer: s.Server, execer: s.Server}
 			err = q.Run(s)
 			if err != nil {
 				return err
 			}
-		case protocol.Describe:
+		case *pgproto3.Describe:
 			err = t.Write(protocol.ErrorResponse(fmt.Errorf("not implemented")))
 			if err != nil {
 				return err
 			}
-		case protocol.Parse:
+		case *pgproto3.Parse:
 			err = t.Write(protocol.ErrorResponse(fmt.Errorf("not implemented")))
 			if err != nil {
 				return err
 			}
-		case protocol.Bind:
+		case *pgproto3.Bind:
 			err = t.Write(protocol.ErrorResponse(fmt.Errorf("not implemented")))
 			if err != nil {
 				return err
 			}
-		case protocol.Execute:
```

Comment:
do we still need all types constants defined in protocol package?

Reply:
no. I'll remove it